libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -178,12 +212,31 @@ bool HTTPClient::beginInternal(String url, const char* expectedProtocol)
 
 bool HTTPClient::begin(String host, uint16_t port, String uri)
 {
+    DEBUG_HTTPCLIENT("[HTTP-Client][begin] host: %s port: %d uri: %s\n", host.c_str(), port, uri.c_str());
     clear();
+
+    // Remove protocol if any
```

Comment:
Makes the function more resilient for me. It is also more consistent with the other overload, making it more easy to swap out the API a developer may want to use.